realm/realm-library/src/androidTest/java/io/realm/TypeBasedNotificationsTests.java

```diff
@@ -501,95 +482,108 @@ public void run() {
         dog.addChangeListener(new RealmChangeListener() {
             @Override
             public void onChange() {
-                assertEquals("Akamaru", dog.getName());
+                // Step 4: Respond to relevant change
                 typebasedCommitInvocations.incrementAndGet();
+                assertEquals("Akamaru", dog.getName());
+                assertEquals(17, dog.getAge());
```

Comment:
call `typebasedCommitInvocations.incrementAndGet();` after assertions, to avoid flaky test.